src/system-info.ts

```diff
@@ -0,0 +1,185 @@
+//               but this state is not sustainable for the long run and might result in throttling if the workload remains the same.
+//               This signal is the last call for the web application to lighten its workload.
+
+export type PressureSource = 'cpu';
+
+export type PressureState = 'nominal' | 'fair' | 'serious' | 'critical';
+
+interface PressureRecord {
+  source: string;
+  state: PressureState;
+  time: number;
+}
+
+interface PressureObserver {
+  observe(source: PressureSource): Promise<void>;
+  unobserve(source: PressureSource): void;
+  disconnect(): void;
+  takeRecords(): PressureRecord[];
+}
+
+declare global {
+  interface Window {
+    PressureObserver?: PressureObserver;
+  }
+}
+
+export enum SystemInfoEvents {
+  CpuPressureStateChange = 'cpu-pressure-state-change',
+}
+
+/**
+ * PressureObserverHelper class to wrap the PressureObserver API
+ * and provide a simple interface for observing pressure state changes.
+ */
+class PressureObserverHelper extends EventEmitter {
+  private observer?: PressureObserver;
+
+  private lastCpuPressure?: PressureState = undefined;
+
+  /**
+   * Creates an instance of PressureObserverHelper.
+   */
+  constructor() {
+    super();
+
+    if (PressureObserverHelper.isPressureObserverSupported()) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      this.observer = new PressureObserver(this.handleStateChange.bind(this));
+      if (this.observer) {
+        this.observer.observe('cpu');
+      }
+    }
+  }
+
+  /**
+   * Handles updates from the PressureObserver.
+   *
+   * @param records - The records from the PressureObserver.
+   */
+  private handleStateChange(records: PressureRecord[]) {
+    records.forEach((record: PressureRecord) => {
+      if (record.source === 'cpu' && record.state !== this.lastCpuPressure) {
+        this.lastCpuPressure = record.state;
+
+        this.emit(SystemInfoEvents.CpuPressureStateChange, record.state);
+      }
+    });
+  }
+
+  /**
+   * Gets the current CPU pressure state.
+   *
+   * @returns The current CPU pressure state, or undefined if API is not supported.
+   */
+  getCpuPressure(): PressureState | undefined {
+    return this.lastCpuPressure;
+  }
+
+  /**
+   * Checks if the Compute Pressure API is supported in the current environment.
+   *
+   * @returns True if the Compute Pressure API is supported, false otherwise.
+   */
+  static isPressureObserverSupported(): boolean {
+    return 'PressureObserver' in window;
+  }
+}
+
+const pressureObserverHelper = new PressureObserverHelper();
+
+/**
+ * SystemInfo class to provide static methods for system information.
+ */
+export class SystemInfo {
+  /**
+   * Checks if the Compute Pressure API is supported in the current environment.
+   *
+   * @returns True if the Compute Pressure API is supported, false otherwise.
+   */
+  static isPressureObserverSupported(): boolean {
```

Comment:
Just a thought I had... every time I look at this API, I wonder to myself how useful exposing it is. But I guess it doesn't hurt to keep it.